bin/imsim.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+"""
+This is the imSim program, used to drive GalSim to simulate the LSST.  Written
+for the DESC collaboration and LSST project.  This version of the program can
+read phoSim instance files as is. It leverages the LSST Sims GalSim interface
+code found in sims_GalSimInterface.
+"""
+from __future__ import absolute_import, print_function
+import os
+import sys
+import logging
+import argparse
+
+from lsst.obs.lsstSim import LsstSimMapper
+
+from lsst.sims.photUtils import LSSTdefaults
+from lsst.sims.utils import ObservationMetaData
+
+from lsst.sims.GalSimInterface.galSimCatalogs import GalSimBase
+from lsst.sims.GalSimInterface import GalSimStars
+from lsst.sims.GalSimInterface import GalSimGalaxies
+
+import desc.imsim
+from desc.imsim.monkeyPatchedGalSimBase import \
+    phoSimCalculateGalSimSeds, phoSimInitilizer, get_phoSimInstanceCatalog
+
+def main(argv):
+    """
+    Drive GalSim to simulate the LSST.
+    """
+    # Monkey Patch the GalSimBase class and replace the routines that
+    # gets the objects and builds the catalog with my version that gets the
+    # information from a phoSim instance catalog file.  This should be temporary
+    # and is a hack.
+    GalSimBase.__init__ = phoSimInitilizer
+    GalSimBase._calculateGalSimSeds = phoSimCalculateGalSimSeds
+    GalSimBase.get_fitsFiles = get_phoSimInstanceCatalog
+
+    # Setup logging output.
+    logging.basicConfig(format="%(message)s", level=logging.INFO,
+                        stream=sys.stdout)
+    logger = logging.getLogger()
+
+    # Setup a parser to take command line arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v', '--version', action='version',
+                        version='%(prog)s 0.0')
+    parser.add_argument('-f', '--file', help="Specify the instance file")
+    parser.add_argument('-n', '--numrows', default=2000, type=int,
+                        help="read the first numrows of the file.")
+    parser.add_argument('--outdir', type=str, default='fits',
+                        help='output directory for eimage file')
+    arguments = parser.parse_args()
+
+    # Get the number of rows to read from the instance file.  Use
+    # default if not specified.
+    numRows = arguments.numrows
+    logger.info("Reading %i rows from the instance catalog %s.",
+                numRows, arguments.file)
+
+    # The PhoSim instance file contains both pointing commands and
```

Comment:
This check:

   # Get the instance file name. A valid file must be specified.
    try:
        with open(arguments.file) as instanceFile:
            print 'Using', instanceFile.name, "as an instance file."
            pass
    except IOError as errorMessage:
        print errorMessage
        print "That instance file cannot be opened. Program will end.\n"
        sys.exit()

was left out from my code.  This means that if you run the program with no arguments it crashes.